pkg/qemu/qemu.go

```diff
@@ -398,7 +430,31 @@ func Cmdline(cfg Config) (string, []string, error) {
 		args = append(args, "-drive", fmt.Sprintf("file=%s,if=virtio", baseDisk))
 	}
 	// cloud-init
-	args = append(args, "-cdrom", filepath.Join(cfg.InstanceDir, filenames.CIDataISO))
+	switch *y.Arch {
+	case limayaml.RISCV64:
+		// -cdrom does not seem recognized for RISCV64
+		args = append(args,
+			"-drive", "id=cdrom0,if=none,format=raw,readonly=on,file="+filepath.Join(cfg.InstanceDir, filenames.CIDataISO),
+			"-device", "virtio-scsi-pci,id=scsi0",
+			"-device", "scsi-cd,bus=scsi0.0,drive=cdrom0")
+	default:
+		// TODO: consider using virtio cdrom for all the architectures
+		args = append(args, "-cdrom", filepath.Join(cfg.InstanceDir, filenames.CIDataISO))
+	}
+
+	// Kernel
+	kernel := filepath.Join(cfg.InstanceDir, filenames.Kernel)
+	kernelCmdline := filepath.Join(cfg.InstanceDir, filenames.KernelCmdline)
+	initrd := filepath.Join(cfg.InstanceDir, filenames.Initrd)
+	if _, err := os.Stat(kernel); err == nil {
+		args = appendArgsIfNoConflict(args, "-kernel", kernel)
+	}
+	if b, err := os.ReadFile(kernelCmdline); err == nil {
+		args = appendArgsIfNoConflict(args, "-append", string(b))
+	}
+	if _, err := os.Stat(initrd); err == nil {
+		args = appendArgsIfNoConflict(args, "-initrd", kernel)
```

Comment:
Shouldn't this be `initrd`?

```suggestion
		args = appendArgsIfNoConflict(args, "-initrd", initrd)
```

Reply:
thanks, fixed